### PR TITLE
bazel/ci: Improve flags/config

### PR DESCRIPTION
**Pull request by phlax**
Currently our bazel flags are not very well structured

- lots of duplicate flags (-> indeterminate outcomes)
- not well setup for both rbe <> local
- not such good separation of responsibility

the problem is 10x worse in mobile, which also hijacks quite a few flags from envoys config

this rinses out duplicates in the config, separates mobile config to its own ~namespace and gets rid of related warnings

this also quietens mobile CI and standardizes the RBE config which will allow adding more complex setups (#28814 )

also removes github mobile token condition that would never be hit and tightens workflow permissions generally

also separates github workflow diskspace cleanup code to an action

and switches default (mostly) for remote exec to `--remote_download_minimal`, updating any ci that requires more downloads, runfiles, etc

more generally moves/consolidates config to the .bazelrc file to make it easier to manage them

broadly, its some first steps to addressing the issues posted above in terms of separating responsibility

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -200,7 +200,9 @@ build:coverage --strategy=CoverageReport=sandboxed,local
 build:coverage --experimental_use_llvm_covmap
 build:coverage --collect_code_coverage
 build:coverage --test_tag_filters=-nocoverage
-build:coverage --instrumentation_filter="//source(?!/common/quic/platform)[/:],//envoy[/:],//contrib(?!/.*/test)[/:]"
+
+build:coverage-envoy --instrumentation_filter="//source(?!/common/quic/platform)[/:],//envoy[/:],//contrib(?!/.*/test)[/:]"
```
Comment (phlax) [inline .bazelrc:204]: this worked previously i think

File: .github/workflows/check-deps.yml [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+  - cron: '0 8 * * *'
   workflow_dispatch:
 
-permissions: read-all
-
 jobs:
   build:
     runs-on: ubuntu-22.04
-    if: |
+    if: >-
```
Comment (phlax) [inline .github/workflows/check-deps.yml:14]: i need to test that this fixes what i think it does

File: ci/do_ci.sh [hunk 3]
```diff
@@ -326,25 +326,10 @@ case $CI_TARGET in
...
-            "--define" "hot_restart=disabled"
-            "--define" "google_grpc=disabled"
-            "--define" "boringssl=fips"
-            "--define" "log_debug_assert_in_release=enabled"
-            "--define" "path_normalization_by_default=true"
-            "--define" "deprecated_features=disabled"
-            "--define" "tcmalloc=gperftools"
-            "--define" "zlib=ng"
-            "--define" "uhv=enabled"
             "--@envoy//bazel:http3=False"
```
Comment (phlax) [inline ci/do_ci.sh:331]: test if these can moved also

File: .github/workflows/check-deps.yml (hunk 2 above)
Comment (phlax) [reply .github/workflows/check-deps.yml:14]: it does

**Comment by phlax** [comment]
this is quite invasive but has been extensively tested, fixes a stack of problems and lays the groundwork to fix a stack more

**Comment by htuch** [comment]
Sending this to @RyanTheOptimist @keith due to the Envoy Mobile implications.

File: .github/workflows/mobile-android_build.yml [hunk 4]
```diff
@@ -165,12 +192,13 @@ jobs:
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (phlax) [inline .github/workflows/mobile-android_build.yml:193]: perhaps this could be empty - the env var would never be normally - ill test

Comment (phlax) [reply .github/workflows/mobile-android_build.yml:193]: tested in staging repo - this is never empty - no matter who opens a PR (and what creds the token has)

**Comment by RyanTheOptimist** [comment]
Does this impact the current ability of failed mobile job to be restarted (which doesn't seem to be available for non-mobile jobs)

**Comment by phlax** [comment]
> Does this impact the current ability of failed mobile job to be restarted (which doesn't seem to be available for non-mobile jobs)

not at all - i havent touched the job handling at all in this PR

File: .github/workflows/POLICY.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ Do not allow any bots or app users to do so, unless this is specifically require
 For example, you could add a `job` condition to prevent any bots from triggering the workflow:
 
 ```yaml
-    if: |
+    if: >-
```
Comment (RyanTheOptimist) [inline .github/workflows/POLICY.md:43]: OOC, what does `>-` mean here?

**Review by RyanTheOptimist — APPROVED** [review]
I think this LGTM. There are quite a few changes, and this code/config can be a bit subtle, but overall it looks like a really nice cleanup.

**Comment by phlax** [comment]
>  There are quite a few changes, and this code/config can be a bit subtle

thanks - there are a few things (eg re configs) that should hopefully become clearer with subsequent prs - i will monitor this closely going through, and have some behind to rebase and push forward also

**Pull request merged**